dev/PushNotifications/PushNotificationManager.h

```diff
@@ -16,7 +16,7 @@ namespace winrt::Microsoft::Windows::PushNotifications::implementation
         static void UnregisterActivator(Microsoft::Windows::PushNotifications::PushNotificationRegistrationActivators const& activators);
         static void UnregisterAllActivators();
 
-        static winrt::Windows::Foundation::IAsyncOperationWithProgress<winrt::Microsoft::Windows::PushNotifications::PushNotificationCreateChannelResult, winrt::Microsoft::Windows::PushNotifications::PushNotificationCreateChannelStatus> CreateChannelAsync(winrt::guid const& remoteId);
+        static winrt::Windows::Foundation::IAsyncOperationWithProgress<winrt::Microsoft::Windows::PushNotifications::PushNotificationCreateChannelResult, winrt::Microsoft::Windows::PushNotifications::PushNotificationCreateChannelStatus> CreateChannelAsync(winrt::guid const remoteId);
```

Comment:
I'm curious that the idl auto generated guid& instead of guid as the header. Is that true and does this impact the projection in any negative way?

Reply:
@kennykerr ?

Reply:
Yeah kenny. Thanks for your response. The above API is a coroutine and the value of remoteId is not held on the stack if the API is suspended. So that is the reason I had to convert this to const value.

Reply:
@kennykerr Should we have c++/winrt generate the pass by value argument automatically for co-routines rather than pass by ref knowing some of the limitations here with argument lifetime?

Reply:
We could, although its probably more reliable to build with code analysis enabled and then you end up with a compiler warning/error:

```C++
winrt::Windows::Foundation::IAsyncOperation<winrt::guid> Async(winrt::guid const& value)
{
    co_await winrt::resume_background();
    co_return value;
}
```

```
main.cpp(5): warning C26811: Lifetime of the memory referenced by parameter ''value'' might end by the time the coroutine is resumed (lifetime.1).
```

![image](https://user-images.githubusercontent.com/9845234/134039168-9fa42148-2892-4923-b04f-12336ebe674d.png)

Reply:
This is good to learn. However, I am wondering which is the right approach here? Enabling code analysis or make the C++/WinRT generate differently for coroutines?

Reply:
C++/WinRT has no idea how you are going to implement the method - you may not use coroutines at all - and once the code is generated all control is lost. However, the C++ compiler can *always* validate that the code is safe regardless of how the code evolves over time. Therefore, having the compiler check the code's correctness seems far more useful.